jenkins/Jenkinsfile

```diff
@@ -0,0 +1,268 @@
+          description: "The bucket where the image will be stored")
+    string(name: "terra_jupyter_base", defaultValue: "0.0.1")
+    string(name: "terra_jupyter_hail", defaultValue: "0.0.1",
+          description: "Is not currently baked into the image, but is here for future functionality")
+    string(name: "terra_jupyter_r", defaultValue: "0.0.3")
+    string(name: "terra_jupyter_python", defaultValue: "0.0.1")
+    string(name: "terra_jupyter_bioconductor", defaultValue: "0.0.2",
+          description: "Is not currently baked into the image, but is here for future functionality")
+    string(name: "leonardo_jupyter", defaultValue: "dev",
+          description: "The tag of the jupyter image to pull in docker image build. Used in the prepare script.")
+    string(name: "welder_server", defaultValue: "latest",
+          description: "The tag of the welder image to pull in docker image build. Used in the prepare script.")
+    string(name: "openidc_proxy", defaultValue: "2.3.1_2",
+          description:"The tag of the openidc proxy image to pull in docker image build. Used in the prepare script.")
+  }
+
+  stages {
+    stage('Terra docker image scan') {
+      steps {
+        git credentialsId: 'jenkins-ssh-github', url: 'git@github.com:DataBiosphere/terra-docker.git', branch: 'master'
+
+        script {
+          def temp = readFile("jenkins/imageDirs.txt").trim().split("\n")
+          images = new ArrayList<String>(Arrays.asList(temp))
+
+          //non terra-docker images
+          images.add('leonardo-jupyter')
+          images.add('welder-server')
+          images.add('openidc-proxy')
+        }
+      }
+    }
+
+    stage('Leonardo Git') {
+      steps {
+        //TODO: FIX BRANCH
+        git credentialsId: 'jenkins-ssh-github', url: 'git@github.com:DataBiosphere/leonardo.git', branch: 'jc-auto-build-dataproc'
+      }
+    }
+        
+    //create short hash properties file
+    stage('Get Short Hash') {
+      steps {
+        script {
+          shortHash = getShortHash("${env.GIT_COMMIT}")
+        }
+      }
+    }
+
+    stage('Docker and Gcloud Auth') {
+      steps {
+        sh """
+          sudo chown -R jenkins:jenkins /home/jenkins/.config
+          docker run --rm  -v /etc/vault-token-dsde:/root/.vault-token:ro broadinstitute/dsde-toolbox:latest vault read --format=json secret/dsde/firecloud/common/image-build-account.json | jq .data > image-build-account.json   
+          gcloud auth activate-service-account --key-file=image-build-account.json
+          gcloud auth configure-docker --quiet
+        """
+      }
+    }
+
+    stage ('Detect Hash Changes') {
+      when { expression { useCustomImages.equals("false") } }
+      steps {
+        script {
+          def boolean shouldUpdate = hasHashChanges(images)
+          shouldAbortBuild = !shouldUpdate //set variable to control if future steps get run
+          def versionMap = getVersionMap(images, "automation/src/test/resources/reference.conf")
```

Comment:
Does this mean it will only pull the images in `automation/src/test/resources/reference.conf`? That file doesn't include welder or the "old" Jupyter image.

Reply:
see line 110